### PR TITLE
1091 http request failed with request info

### PR DESCRIPTION
**Pull request by hui-an-yang**
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito-http-utils/src/taquito-http-utils.ts [hunk 1]
```diff
@@ -49,10 +49,14 @@ export class HttpResponseError extends Error {
  *  @description Error that indicates a general failure in making the HTTP request
  */
 export class HttpRequestFailed extends Error {
-  public name = 'HttpRequestFailed';
+  public name = 'HttpRequest';
```
Comment (zainen) [inline packages/taquito-http-utils/src/taquito-http-utils.ts:52]: `HttpRequestFailed` might be a little more clear

**Review by zainen — COMMENTED** [review]
nice! 😄

File: packages/taquito-http-utils/src/taquito-http-utils.ts [hunk 2]
```diff
@@ -49,10 +49,14 @@ export class HttpResponseError extends Error {
  *  @description Error that indicates a general failure in making the HTTP request
  */
 export class HttpRequestFailed extends Error {
-  public name = 'HttpRequestFailed';
+  public name = 'HttpRequesFailed';
```
Comment (roxaneletourneau) [inline packages/taquito-http-utils/src/taquito-http-utils.ts:52]: There is a "t" missing

File: integration-tests/http-backend-request-error-handling.spec.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { HttpBackend } from '@taquito/http-utils';
+
+describe('HttpBackend request', () => {
+    it('will fail with http request url and request method', async (done) => {
+        try {
+            const http: HttpBackend = new HttpBackend(1);
+            await http.createRequest<string>({
+              method: 'GET',
+              url: 'https://mainnet.api.tez.ie/chains/main/blocks/head/hash'
```
Comment (roxaneletourneau) [inline integration-tests/http-backend-request-error-handling.spec.ts:9]: Could you also add an expect statement for the message to make sure it renders properly?

File: packages/taquito-http-utils/src/taquito-http-utils.ts (hunk 2 above)
Comment (hui-an-yang) [reply packages/taquito-http-utils/src/taquito-http-utils.ts:52]: sorry, added it back

File: integration-tests/http-backend-request-error-handling.spec.ts (hunk 3 above)
Comment (hui-an-yang) [reply integration-tests/http-backend-request-error-handling.spec.ts:9]: yes, added expect statement with message and pushed another commit

**Review by roxaneletourneau — APPROVED** [review]
LGTM!
Can be merged when the integration tests in the pipeline pass :)

**Pull request merged**